stdlib/weakref.pyi

```diff
@@ -42,6 +42,7 @@ ProxyTypes: tuple[type[Any], ...]
 
 class WeakMethod(ref[_CallableT]):
     def __new__(cls, meth: _CallableT, callback: Callable[[Self], object] | None = None) -> Self: ...
+    def __init__(self, meth: _CallableT, callback: Callable[[Self], object] | None = None) -> None: ...
```

Comment:
I'm not sure we should define `WeakMethod.__init__` as well. `WeakMethod` just inherits `ReferenceType.__init__`, and trying to invoke it using the renamed arguments fails:

```python
>>> class Foo:
...     def bar(): ...
... 
>>> from weakref import WeakMethod
>>> WeakMethod(meth=Foo().bar)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: ref() takes no keyword arguments
>>> WeakMethod(Foo().bar)
<weakref at 0x7ce4afbf3dd0; dead>
```

Reply:
Ah, I see, makes sense, let me revert this.

Reply:
Removed it. However, to avoid Pyright complaining I had to consolidate the types a bit, it seems like we just want the parent class accepting `Self` in `callback`, and the other args (guessing it was maybe type hinted before `Self?`), and enforce positional arguments in `new`.

Reply:
Wasn't sure on `Any` vs `object` though for the return type though.